### PR TITLE
Add support for Django 3.0, drop Django 1.x, Python 2

### LINKED ISSUE #382: Django 3.0 Compatibility
I just upgraded to Django 3.0 and this failed for my project with the following exception:

```
Traceback (most recent call last):
  File "/Users/dneuhaeuser/dev/pricing-steering/manage.py", line 21, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/site-packages/django/core/management/__init__.py", line 377, in execute
    django.setup()
  File "/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/site-packages/django/apps/registry.py", line 91, in populate
    app_config = AppConfig.create(entry)
  File "/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/site-packages/django/apps/config.py", line 90, in create
    module = import_module(entry)
  File "/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/site-packages/django_rq/__init__.py", line 3, in <module>
    from .decorators import job
  File "/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/site-packages/django_rq/decorators.py", line 4, in <module>
    from django.utils import six
ImportError: cannot import name 'six' from 'django.utils' (/Users/dneuhaeuser/dev/pricing-steering/.tox/py37/lib/python3.7/site-packages/django/utils/__init__.py)
```

This looks like django-rq isn't compatible with Django 3.0, yet. It would be great, if that could be fixed.

### PR DESCRIPTION
**Pull request by hugorodgerbrown**
Fixes #382 

---

### Summary

* Add support for Django 3.0
* Drop support for Django 1.x
* Drop support for Python 2.x
* Add Python 3.8 to test matrix

### Version support 

| Django | 1.x | 2.x | 3.x |
| :-- | :-- | --- | --- |
| Python 2.7 | ✖️  | ✖️ | ✖️ | 
| Python 3.5 | ✖️ | 👍| ✖️^  |
| Python 3.6 | ✖️ | 👍| 👍 |
| Python 3.7 | ✖️ | 👍| 👍|
| Python 3.8 | ✖️ | 👍| 👍|

^ Django 3.0 does not support Python 3.5 (https://docs.djangoproject.com/en/3.0/faq/install/#what-python-version-can-i-use-with-django)

### Changes

Aside from supporting Django 3.0, the dropping of Django 1.x and Python 2.x enables a number of changes across the project, including:

* Replace `django.conf.urls` with `django.urls` (and use `path`, `re_path` instead of `url`)
```diff
-from django.conf.urls import url
+from django.urls import path
...
 urlpatterns = [
-    url(r'^$', views.home, name='home'),
-    url(r'^admin/', admin.site.urls),
+    path('', views.home, name='home'),
+    path('admin/', admin.site.urls),
 ]
```
* Use of `unittest.mock` instead of importing `mock` project
* Replace `admin_static` template tags with `static`

### Additional changes (open to debate):

1. Consolidate all local imports as relative
2. Add a Pipfile for aiding with local development

### CODE REVIEW

**Comment by hugorodgerbrown** [comment]
Looks like `django-redis` needs a bump as well. Will investigate.

**Comment by hugorodgerbrown** [comment]
Blocked on https://github.com/niwinz/django-redis/pull/394

**Comment by hugorodgerbrown** [comment]
Which is in turn blocked on https://github.com/niwinz/django-redis/pull/387

Turtles a long way down on this one.

**Comment by selwin** [comment]
I think we should drop Python 2 support and support Django >= 2.0. Could you please get rid of `six` altogether?

**Comment by hugorodgerbrown** [comment]
@selwin since #385 and #384 are solving the same issue, I think I should close this in favour of that PR as it's further ahead?

**Comment by selwin** [comment]
Yes, thanks. Any help in getting the other PR in will be appreciated 😀

**Comment by hugorodgerbrown** [comment]
Closing in favour of #385

**Comment by hugorodgerbrown** [comment]
@selwin Couldn't help it. Have pulled in the changes from #385 and re-opened. There is a problem with one of the tests under Python 3.5 - not sure how to fix that one.

**Comment by hugorodgerbrown** [comment]
Have also added a `Pipfile` - happy to remove that if not appropriate, but it helped when trying to get the tests running locally.

File: .travis.yml [hunk 1]
```diff
@@ -2,39 +2,26 @@ language: python
...
-    - python: "3.7"
-      env: DJANGO=1.11.16
+    - python: "3.5"
+      env: DJANGO=3.0
 
 install:
-  - pip install Django==$DJANGO docutils pygments rq-scheduler django-redis
+  - pip install Django==$DJANGO docutils pygments rq-scheduler
+  # django-redis fix not yet released, so pinning to master
+  - pip install git+https://github.com/niwinz/django-redis.git@13acf21d61ec95de6550c2df9fc65363676cf8a7#egg=django_redis
```
Comment (hugorodgerbrown) [inline .travis.yml]: `django-redis` pinned to a commit that support Django3.

File: django_rq/django_rq [hunk 2]
```diff
@@ -1 +0,0 @@
-django_rq
```
Comment (hugorodgerbrown) [inline django_rq/django_rq]: This file was causing import issues, and I can't work out what it was in the project for. If anyone can shed any light on that will replace?

File: Pipfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[[source]]
```
Comment (hugorodgerbrown) [inline Pipfile]: Added to making hacking on the project easier (for people with `pipenv` installed) - happy to remove.

File: django_rq/admin.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 from django.contrib import admin
-from django_rq import settings
+
+from . import settings
```
Comment (hugorodgerbrown) [inline django_rq/admin.py]: There was a mix of relative and absolute imports within `django_rq` itself - I've made all internal package import relative, although I have left imports from `django_rq.tests` to `django_rq.*` as absolute?

File: django_rq/urls.py [hunk 5]
```diff
@@ -1,12 +1,7 @@
-from django_rq import views
+from django.urls import re_path
```
Comment (hugorodgerbrown) [inline django_rq/urls.py]: I've updated the urls to the newer format.

**Comment by selwin** [comment]
Thanks!

**Pull request merged**